From ee5be6bc4fec8abc88289059cfd99b456b0904e9 Mon Sep 17 00:00:00 2001
Message-ID: <ee5be6bc4fec8abc88289059cfd99b456b0904e9.1703031694.git.sharlatanus@gmail.com>
In-Reply-To: <cover.1703031693.git.sharlatanus@gmail.com>
References: <cover.1703031693.git.sharlatanus@gmail.com>
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Mon, 18 Dec 2023 13:42:52 +0000
Subject: [PATCH 25/32] gnu: python-stsci-imagestats: Enable tests

* gnu/packages/astronomy.scm (python-stsci-imagestats): Enable tests.
[arguments]<#:phases>: Add 'build-extensions phase.
[native-inputs]: Add python-pytest.

Change-Id: I8cb35fd083453d577eb6ad94f097502566c1bcee
---
 gnu/packages/astronomy.scm | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/gnu/packages/astronomy.scm b/gnu/packages/astronomy.scm
index 00653efd70..1e4d54b2d4 100644
--- a/gnu/packages/astronomy.scm
+++ b/gnu/packages/astronomy.scm
@@ -2690,9 +2690,15 @@ (define-public python-stsci-imagestats
                 "01jqqrhcna0ghin48bbcza57d12371ny4l4pqws89irwdhd8xr0r"))))
     (build-system pyproject-build-system)
     (arguments
-     (list #:tests? #f)) ;No tests
+     (list
+      #:phases
+      #~(modify-phases %standard-phases
+          (add-before 'check 'build-extensions
+            (lambda _
+              ;; Cython extensions have to be built before running the tests.
+              (invoke "python" "setup.py" "build_ext" "--inplace"))))))
     (propagated-inputs (list python-numpy))
-    (native-inputs (list python-setuptools-scm))
+    (native-inputs (list python-pytest python-setuptools-scm))
     (home-page "https://stsciimagestats.readthedocs.io/en/latest/")
     (synopsis "Compute sigma-clipped statistics on data arrays")
     (description
-- 
2.41.0

